### PR TITLE
OTP Geocoder Clustering in UI

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
new OTP changes (https://github.com/opentripplanner/OpenTripPlanner/pull/5879) will support making geocoder clustering visible to users. This PR exposes this into the location field.

Breaking change for Geocoder! Old OTP versions are no longer supported

### CODE REVIEW

**Review by binh-dam-ibigroup — APPROVED** [review]
Mostly styles and type refinements, otherwise looks good.

File: packages/geocoder/src/apis/otp/index.ts [hunk 1]
```diff
@@ -7,18 +7,24 @@ type FetchArgs = {
...
+  name: string,
+  id: string,
+  type: "STOP" | "STATION"
+  agencies?: { id: string, name: string }[]
+  feedPublisher?: { name: string }
+  modes: string[] 
+}
+
 type OTPGeocoderResponse = {
   results: {
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/geocoder/src/apis/otp/index.ts:25]: results will never be null

File: packages/geocoder/src/geocoders/otp.ts [hunk 2]
```diff
@@ -45,16 +45,19 @@ export default class OTPGeocoder extends Geocoder {
   rewriteAutocompleteResponse(response: OTPGeocoderResponse): MultiGeocoderResponse {
     return {
       features: response?.results?.map(stop => ({
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/geocoder/src/geocoders/otp.ts:47]: We needed `stop` for the label generation but I've still cleaned it up!

File: packages/location-field/src/styled.ts [hunk 3]
```diff
@@ -128,6 +128,12 @@ export const OptionSubTitle = styled.span`
   margin-left: 6px;
 `;
 
+export const OptionAltLabel = styled.div`
+  color: #686868;
+  font-size: 11px;
+  margin-left: 46px; /* This is a bit brittle. TODO: Calculate */
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/location-field/src/styled.ts:134]: yup

**Review by josh-willis-arcadis — APPROVED** [review]
LGTM

**Pull request merged**